numpy/lib/npyio.py

```diff
@@ -1952,7 +1954,8 @@ def genfromtxt(fname, dtype=float, comments='#', delimiter=None,
                     if i in user_converters:
                         ishomogeneous &= (ttype == dtype.type)
                         if ttype == np.string_:
-                            ttype = "|S%i" % max(len(row[i]) for row in data)
+                            ttype = "|S%i" % (1 if not data else
+                                              max(len(row[i]) for row in data))
```

Comment:
Better as the following, I think:

```
maxlen = 1 if not data else max(len(row[i]) for row in data)
ttype = (np.string_, maxlen)
```

Reply:
What is this change trying to fix? Sounds like it might just be a workaround for #8977

Reply:
This is the test with the converter `{0: lambda arg: arg}` we end up in this code branch because the automatically detected dtype is string I think, then you have a `ValueError: max() arg is an empty sequence`.